goal_src/engine/camera/cam-update.gc

```diff
@@ -5,3 +5,735 @@
 ;; name in dgo: cam-update
 ;; dgos: GAME, ENGINE
 
+;; definition for function plane-from-points
+;; WARN: Unsupported inline assembly instruction kind - [mula.s f0, f3]
+;; WARN: Unsupported inline assembly instruction kind - [madda.s f1, f4]
+;; WARN: Unsupported inline assembly instruction kind - [madd.s f0, f2, f5]
+(defun
+  plane-from-points
+  ((arg0 (inline-array plane))
+   (arg1 vector)
+   (arg2 vector)
+   (arg3 vector)
+   (arg4 int)
+   )
+  (local-vars (f0-1 float))
+  (rlet ((acc :class vf)
+         (vf1 :class vf)
+         (vf2 :class vf)
+         (vf3 :class vf)
+         )
+   (let ((s4-0 (new 'stack-no-clear 'vector)))
+    (let ((v1-0 s4-0))
+     (let ((a1-1 arg2))
+      (.lvf vf1 (&-> arg1 quad))
+      (.lvf vf2 (&-> a1-1 quad))
+      )
+     (.outer.product.vf vf3 vf1 vf2)
+     (.svf (&-> v1-0 quad) vf3)
+     )
+    (vector-normalize! s4-0 1.0)
+    (let* ((v1-1 s4-0))
+     (set! f0-1 (vector-dot v1-1 arg3))
```

Comment:
This is fine, but the decompiler could make it look nicer (https://github.com/water111/jak-project/issues/636 )